benchmarks/spack/repo/packages/sphng/package.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Sphng(MakefilePackage):
+    """sphNG benchmark for DiRAC.
+
+    The source code for this benchmark is stored in a private repository. To
+    gain access please contact the RSE team at the University of Leicester or
+    contact via github from our organization page
+    https://github.com/UniOfLeicester
+    """
+
+    homepage = "https://github.com/UniOfLeicester/benchmark-sphng"
+    git = "ssh://git@github.com/UniOfLeicester/benchmark-sphng.git"
+
+    maintainers = ["TomMelt"]
+
+    version("v1.0.0", tag="v1.0.0")
+
+    executables = [r"^sph_tree_rk_gradh$"]
+
+    depends_on("mpi")
+
+    parallel = False
+
+    def edit(self, spec, prefix):
+        self.fc = spack_fc if "~mpi" in spec else spec["mpi"].mpifc
+
+        env["PREFIX"] = prefix
+        env["SYSTEM"] = "SPACK"
+        env["mpi"] = "yes"
+        env["openmp"] = "yes"
+
+        env["FC"] = self.fc
+        env["OMPFLAG"] = self.compiler.openmp_flag
+        if self.compiler.name == "intel":
+            fflags = (
+                "-O3 -mavx2 -mfma -mcmodel=medium -warn uninitialized -warn truncated_source "
+                "-warn interfaces -nogen-interfaces -DINCMPI"
+            )
+            env["DBLFLAG"] = "-r8"
+            env["DEBUGFLAG"] = "-check all -traceback -g -fpe0 -fp-stack-check -heap-arrays -O0"
+            env["ENDIANFLAGBIG"] = "-convert big_endian"
+            env["ENDIANFLAGLITTLE"] = "-convert little_endian"
+        elif self.compiler.name == "gcc":
+            fflags = (
+                "-m64 -mcmodel=large -O3 -I. -Wall -Wno-conversion -Wno-unused-dummy-argument "
+                "-Wno-maybe-uninitialized -Warray-temporaries"
+            )
+            env["DBLFLAG"] = "-fdefault-real-8 -fdefault-double-8"
+            env["DEBUGFLAG"] = "-g"
+            env["ENDIANFLAGBIG"] = "-fconvert=big-endian"
+            env["ENDIANFLAGLITTLE"] = "-fconvert=little-endian"
```

Comment:
I played with building this using gcc and I ran into this error
```
densityiterate_gradh_MPI.F:2028:22:

 2028 |  29               IF (ifound.EQ.0) THEN
      |                      1
Error: Operands of comparison operator '.eq.' at (1) are LOGICAL(4)/INTEGER(4)
```
As far as I can tell, comparing integers and logicals in if statements is [not allowed by gfortran](https://gcc.gnu.org/onlinedocs/gfortran/Implicitly-convert-LOGICAL-and-INTEGER-values.html) so unless we want to touch the source code maybe just drop `gcc` from supported compilers?

I also had to add ` -fallow-argument-mismatch` to `fflags` to even get that far

Reply:
Yeah, I agree. It requires code modification therefore we should just remove gcc support.